vyper/parser/external_call.py

```diff
@@ -68,6 +69,15 @@ def external_contract_call(node,
         ['assert', ['extcodesize', contract_address]],
         ['assert', ['ne', 'address', contract_address]],
     ]
+    if context.is_constant and not sig.const:
+        raise ConstancyViolationException(
+            "May not call non-constant function '%s' within %s." % (
```

Comment:
Add suggested use case here perhaps as well?
ie. "for checking returns of modifiable contract calls check ret: bool = self.Modify()
assert ret"